### PR TITLE
[LE] Wire in the metrics from client to server

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
We have implemented the calculation and aggregating of leader election time, but not yet wired it so that the client sends to the server.

**Implementation Description (bullets)**:
Tacked the sending of metrics on to the current timelock feedback background task. 

**Testing (What was existing testing like?  What have you done to improve it?)**:
Didn't add any testing, there was none existing.

**Concerns (what feedback would you like?)**:
I think this was what we intended.

**Where should we start reviewing?**:
`TimeLockFeedbackBackgroundTask`

**Priority (whenever / two weeks / yesterday)**:
Want to get this through quickly.

### CODE REVIEW

File: lock-api/src/main/java/com/palantir/lock/client/LeaderElectionReportingTimelockService.java [hunk 1]
```diff
@@ -63,7 +63,8 @@
     private Map<UUID, Instant> leadershipUpperBound = new ConcurrentHashMap<>();
     private Map<UUID, Instant> leadershipLowerBound = new ConcurrentHashMap<>();
 
-    public LeaderElectionReportingTimelockService(
+    @VisibleForTesting
+    LeaderElectionReportingTimelockService(
```
Comment (Jolyon-S) [inline lock-api/src/main/java/com/palantir/lock/client/LeaderElectionReportingTimelockService.java:67]: Changes to this class are just cleanup (adding static, moving methods around).

File: lock-api-objects/build.gradle [hunk 2]
```diff
@@ -27,7 +27,7 @@ recommendedProductDependencies {
     productDependency {
         productGroup = 'com.palantir.timelock'
         productName = 'timelock-server'
-        minimumVersion = '0.218.0'
+        minimumVersion = '0.313.0'
```
Comment (jeremyk-91) [inline lock-api-objects/build.gradle:30]: This has AtlasDB 0.265.0, which is necessary.

File: lock-api/src/main/java/com/palantir/lock/client/LeaderElectionReportingTimelockService.java [hunk 3]
```diff
@@ -230,26 +232,30 @@ LeaderElectionStatistics getStatisticsAndSetRegistryTo(TaggedMetricRegistry metr
         return durationToNextLeader(lowerBounds, upperBounds, leaders, secondToLastLongTermLeader);
     }
 
-    private Set<UUID> leadersWithBothBounds(Map<UUID, Instant> lowerBounds, Map<UUID, Instant> upperBounds) {
+    private void clearOldLongTermLeaders(List<UUID> sortedLongTermLeaders) {
+        for (int i = 0; i < sortedLongTermLeaders.size() - 2; i++) {
```
Comment (jeremyk-91) [inline lock-api/src/main/java/com/palantir/lock/client/LeaderElectionReportingTimelockService.java:236]: this is a bit trippy, but correct

**Review by jeremyk-91 — APPROVED** [review]
🕐  0:08 🏁  0:16

Looks good!

**Review by Jolyon-S — COMMENTED** [review]
:+1:
